src/data/zsqrtd/to_real.lean

```diff
@@ -0,0 +1,45 @@
+/-
+Copyright (c) 2021 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mario Carneiro
+-/
+import data.real.sqrt
+import data.zsqrtd.basic
+
+/-!
+# Image of `zsqrtd` in `ℝ`
+
+This file defines `zsqrtd.to_real` and related lemmas.
+It is in a separate file to avoid pulling in all of `data.real` into `data.zsqrtd`.
+-/
+
+namespace zsqrtd
+
+/-- The image of `zsqrtd` in `ℝ`, using `real.sqrt` which takes the positive root of `d`.
+
+If the negative root is desired, use `to_real h a.conj`. -/
+@[simps]
+noncomputable def to_real {d : ℤ} (h : 0 ≤ d) : ℤ√d →+* ℝ := {
```

Comment:
I think it will be useful to generalise this to a `lift` that takes an element `x : R` (for some ring `R`) such that `x^2 = d`.
Could you please do that?

It's fine to also specialise to a `to_real` that chooses the positive root of `d` in `real`.

Reply:
Good idea, done. I couldn't work out how to generalize `to_real_injective` - I assume I'm relying on the injectivity of the map from int to reals, but I couldn't work out where, and `show_term` wasn't revealing to me the relevant lemma.